### PR TITLE
ListSets and multiple abstract/description elements bug fixes

### PR DESCRIPTION
**Pull request by jefferya**
Fixes:

1. Multiple abstract and description in oai_dc and oai_etdms metadata #12 

    * Handle the case where a Fedora object has multiple values for a single dcterms:abstract or dcterms:description property. This occurs in the context of Fedora 4.2 & HydraNorth when a sufficiently long description or abstract is added triggering a change in property type to `Binary` from `String` and subsequent changes to the contents of the abstract or description add a new value instead of replacing the current.

    * Workaround: OAI-Provider uses the last value element associated with the property.

2. Not all sets included in the ListSets response #15

### CODE REVIEW

File: src/main/java/org/fcrepo/oai/generator/JcrOaiEtdmsGenerator.java [hunk 1]
```diff
@@ -214,8 +214,22 @@ public Thesis generate(final Session session, final Container obj, final String
...
+               * if dcterms:description property (in jcr/xml) is of a
+               * substantial length (tested with 6760 characters) the property
+               * type switches to `Binary` with subsequent edits appending a new
+               * value to the list of values for the property (i.e., not replacing)
+               * This workaround chooses the last value assuming the previous
+               * values are old versions (to avoid returning all values of the
+               * property, including the obsolete). 2017-05-12
+               */
+                Value lastValue = null;
                 for (final Value v : prop.getValues()) {
```
Comment (piyapongch) [inline src/main/java/org/fcrepo/oai/generator/JcrOaiEtdmsGenerator.java]: Can we do something like this?
Value lastValue = prop.getLenght() > 0 ? prop.getValues()[prop.getLength()-1] : null;
This should to the last value without looping through all of them to get the last value.

Comment (jefferya) [inline src/main/java/org/fcrepo/oai/generator/JcrOaiEtdmsGenerator.java]: I like this idea @piyapongch - should work and might improve performance. Thanks!

Comment (jefferya) [inline src/main/java/org/fcrepo/oai/generator/JcrOaiEtdmsGenerator.java]: @piyapongch 

prop.getLength() returns the length of the value (not the length of the array of values). Trying another way.

```
final Value[] vals = prop.getValues();
final int len = java.lang.Math.toIntExact(vals.length);
if (len > 0) {
    final Value lastValue = (len > 0) ? vals[len - 1] : null;
    addDescription(lastValue, thesis.getDescription(), prefix);
    }
```

**Comment by jefferya** [comment]
Local machine tests pass. Suggestions from @piyapongch incorporated and ready for review. After that, release building for the test server.

**Pull request merged**